src/data/list/cycle.lean

```diff
@@ -606,4 +628,7 @@ by { rw [←next_reverse_eq_prev, ←mem_reverse_iff], exact next_mem _ _ _ _ }
 
 end decidable
 
+instance [has_repr α] : has_repr (cycle α) :=
+⟨λ s, "{" ++ string.intercalate ", " ((s.map repr).lists.sort (≤)).head ++ "}"⟩
```

Comment:
I don't think this is a good idea unless you also introduce notation to go in the other direction.

Reply:
Changed to be more like the notation I plan on adding in a follow-up PR.